connector_prestashop/models/product_template/common.py

```diff
@@ -118,11 +118,28 @@ def recompute_prestashop_qty(self):
         return True
 
     def _prestashop_qty(self):
+        root_location = (self.backend_id.stock_location_id or
+                         self.backend_id.warehouse_id.lot_stock_id)
         locations = self.env['stock.location'].search([
-            ('id', 'child_of', self.backend_id.warehouse_id.lot_stock_id.id),
+            ('id', 'child_of', root_location.id),
             ('prestashop_synchronized', '=', True),
             ('usage', '=', 'internal'),
         ])
+        # if we choosed a location but none where flagged
+        # 'prestashop_synchronized', consider we want all of them in the tree
+        if not locations:
+            locations = self.env['stock.location'].search([
+                ('id', 'child_of', root_location.id),
+                ('usage', '=', 'internal'),
+            ])
+        if not locations:
+            # we must not pass an empty location or we would have the
+            # stock for every warehouse, which is the last thing we
+            # expect
+            raise exceptions.UserError(
```

Comment:
But is this going to be silently thrown to the log only or is this reflected on a connector job?